envoy/router/context.h

```diff
@@ -23,6 +24,11 @@ class Context {
    */
   virtual const VirtualClusterStatNames& virtualClusterStatNames() const PURE;
 
+  /**
+   * @return a struct containing StatNames for path level stats.
```

Comment:
Instead of "path" I would call this "route" level stats as they might match on a lot of different things. Change this here and elsewhere.